### PR TITLE
Count_Inversion.java

### LINKED ISSUE #1664: Count inversion in c++ and java.
The code in the repo are given in c and js. Please assgin me this for c++ and java.

### PR DESCRIPTION
**Pull request by mansikagrawal**
Fixes #1664 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:Added file Count_Inversion.java
- 
- 

#### Languages Used:Java
- 
- 

#### Files Added:Count_Inversion.java
- 
-

### CODE REVIEW

**Comment by somya-kapoor** [comment]
No indentation present in the code.

**Comment by mansikagrawal** [comment]
@somya-kapoor  MaM please check now

File: Count_Inversion/Count_Inversion.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+   static int sortfn( int arr[], int n ) 
+    { 
+      int temp[] = new int[n]; 
+      return mergeSort(arr, temp, 0, n - 1); 
+    } 
+
+    /* This function returns  the number of inversions in the array */
+    static int mergeSort( int arr[], int temp[], int left, int right ) 
+    { 
+     int mid, invcount = 0;
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: Please correct the indentation.

File: Count_Inversion/Count_Inversion.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      return mergeSort(arr, temp, 0, n - 1); 
+    } 
+
+    /* This function returns  the number of inversions in the array */
+    static int mergeSort( int arr[], int temp[], int left, int right ) 
+    { 
+     int mid, invcount = 0; 
+     if (right > left) 
+     { 
+      /* Divide the array into two parts */
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: Please correct the indentation.

File: Count_Inversion/Count_Inversion.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      /* inversions in the right subarray */
+      invcount += mergeSort( arr, temp, mid + 1, right); 
+
+      /* merge counts the number of cross inversions */
+      invcount += mergefn( arr, temp, left, mid + 1, right); 
+     } 
+     return invcount; 
+   } 
+
+    /* This method merges two sorted arrays and returns inversion count in
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: Please correct the indentation.

**Comment by somya-kapoor** [comment]
@mansikagrawal Please correct the indentation.

**Comment by somya-kapoor** [comment]
@mansikagrawal Are you working on it?

**Comment by mansikagrawal** [comment]
@somya-kapoor  Mam please check

File: Count_Inversion/Count_Inversion.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
+//Count Inversion in java
+
+import java.util.*;
+class CountInversion
+ {
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: inconsistency in indentation

File: Count_Inversion/Count_Inversion.java (hunk 1 above)
Comment (somya-kapoor) [reply Count_Inversion/Count_Inversion.java]: inconsistency in indentation

File: Count_Inversion/Count_Inversion.java (hunk 2 above)
Comment (somya-kapoor) [reply Count_Inversion/Count_Inversion.java]: inconsistency in indentation

File: Count_Inversion/Count_Inversion.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        while ( i <= mid - 1 ) 
+        temp[k++] = arr[i++]; 
+
+        /* Copy the remaining elements of right subarray to temp*/
+        while ( j <= right ) 
+        temp[k++] = arr[j++]; 
+
+        /*Copy back the merged elements to original array*/
+        for ( i = left; i <= right; i++ ) 
+        arr[i] = temp[i];
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: inconsistency in indentation

File: Count_Inversion/Count_Inversion.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+           } 
+      } 
+
+        /* Copy the remaining elements of left subarray to temp*/
+        while ( i <= mid - 1 ) 
+        temp[k++] = arr[i++]; 
+
+        /* Copy the remaining elements of right subarray to temp*/
+        while ( j <= right ) 
+        temp[k++] = arr[j++];
```
Comment (somya-kapoor) [inline Count_Inversion/Count_Inversion.java]: inconsistency in indentation

**Pull request closed without merging**